src/OpenTelemetry.Instrumentation.Http/HttpClientTraceInstrumentationOptions.cs

```diff
@@ -125,6 +125,9 @@ public class HttpClientTraceInstrumentationOptions
     /// </remarks>
     public bool RecordException { get; set; }
 
+    internal List<string> KnownHttpMethods { get; set; } =
```

Comment:
Is this being set from `OTEL_INSTRUMENTATION_HTTP_KNOWN_METHODS` anywhere? I don't see a ctor accepting `IConfiguration` on this class so it is possible some prep works needs to be done to hook that up. LMK if you would like some help with that.

Reply:
Correct, that was still a to do.
I now made some changes to read from env variables as well and added a unit test for it.
Can you review?

Reply:
A question I had:
When `OTEL_INSTRUMENTATION_HTTP_KNOWN_METHODS` is set to an empty string or something illogical like `fooBar`,  `http.request.method` is still set to `GET` for example and not to `_OTHER`.
Do you think this is correct behaviour?

Reply:
I pushed some changes to the branch. Sorry about that I needed to see how the code was working and then I just started messing around with it 😓

* Refactored everything into `RequestMethodHelper` (instead of options classes). Reason for that is `OTEL_INSTRUMENTATION_HTTP_KNOWN_METHODS` should also apply to metrics. I didn't want to introduce metrics options classes where we don't have them and also didn't want the same code in 4 options classes 🤣 I think now it has better separation of concerns.

* Your question above about `fooBar`. I took a look at the spec and it seems like `OTEL_INSTRUMENTATION_HTTP_KNOWN_METHODS` should fully override the default list. The way it was coded you could only pair down the list of known things. I switched it up so we respect _whatever_ is specified by `OTEL_INSTRUMENTATION_HTTP_KNOWN_METHODS`. I would like @vishweshbankwar to sound off on this. He has more expertise in the area. But he is out until 1/15 so we'll have to wait a bit to get his review.